pkg/cache/application.go

```diff
@@ -568,22 +581,6 @@ func (app *Application) handleCompleteApplicationEvent() {
 	}()
 }
 
-func failTaskPodWithReasonAndMsg(task *Task, reason string, msg string) {
-	podCopy := task.GetTaskPod().DeepCopy()
-	podCopy.Status = v1.PodStatus{
```

Comment:
Moved failTaskPodWithReasonAndMsg() to task.go



change
- podCopy := task.GetTaskPod().DeepCopy()
to
- podCopy := task.pod.DeepCopy()

to prevent deadlock when task state machine is handling TaskRejected event. (WLock)